dts/bindings/pinctrl/infineon,xmc4xxx-pinctrl.yaml

```diff
@@ -0,0 +1,117 @@
+
+  To create a new pin configuration, the user may append to the &pinctrl node, for example
+
+  #include <zephyr/dt-bindings/pinctrl/xmc4xxx-pinctrl.h>
+  &pinctrl {
+      my_node_config: my_node_config {
+      pinmux = <XMC4XXX_PINMUX_SET(0, 1, 2)>;
+      drive-push-pull;
+        ... other supported pin configurations ..
+  };
+  where XMC4XXX_PINMUX_SET(PORT, PIN, ALTERNATE_FUNCTION) is a helper macro for setting the
+  alternate function for a given port/pin. Setting ALTERNATE_FUNCTION = 0 means that no
+  alternate function is selected.
+
+  The pinctrl driver only sets the alternate function for output pins. The input mux is
+  handled by the peripheral drivers. For example the &usic1ch1 node has input-src property for
+  this purpose. There are no pre-defined nodes for the input mux and this must be properly set
+  by the user. Refer to the peripheral .yaml file (i.e. infineon,xmc4xxx-uart.yaml) and
+  XMC4XXX documentation.
+
+compatible: "infineon,xmc4xxx-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node.yaml
+      child-binding:
+        property-allowlist:
+          - bias-pull-down
+          - bias-pull-up
+          - drive-push-pull
+          - drive-open-drain
+          - output-high
+          - output-low
+
+properties:
+    "#address-cells":
+      required: true
+      const: 1
+    "#size-cells":
+      required: true
+      const: 1
+
+child-binding:
+    description: |
+      Each child node defines the configuration for a particular state.
+
+    properties:
+        pinmux:
+          description: |
+            Encodes port/pin and alternate function. See helper macro XMC4XX_PINMUX_SET().
+            Alternate function is only set for output pins; It selects  ALT1-ALT4
+            output line in the GPIO element. The alternate function for input pins is
+            handled separately by the peripheral. It is upto the peripheral to configure which
+            input pin to use (For example see parameter input-src in infineon,xmc4xxx-uart.yaml).
+          required: true
+          type: int
+
+        drive-strength:
+          description: |
+            Drive strength of the output pin. Following options as in XMC_GPIO_OUTPUT_STRENGTH
+            See xmc4_gpio.h. This only has an effect if the pin is in drive-push-pull mode.
+          required: false
+          type: string
+          default: "strong-medium-edge"
+          enum:
+            - "strong-sharp-edge"
+            - "strong-medium-edge"
+            - "strong-soft-edge"
+            - "strong-slow-edge"
+            - "medium"
+            - "medium-unknown1-edge"
+            - "medium-unknown2-edge"
+            - "weak"
+
+        invert-input:
+          description: |
+            Inverts the input.
+          required: false
+          type: boolean
+
+        hwctrl:
+          description:
+            Pre-assigns hardware control of the pin to a certain peripheral.
```

Comment:
https://docs.zephyrproject.org/latest/build/dts/bindings.html#general-rules says:

If your description is short, it’s fine to use this style:

description: my short string

If your description is long or spans multiple lines, you must use this style:

description: |
  My very long string
  goes here.
  Look at all these lines!

This | style prevents YAML parsers from removing the newlines in multi-line descriptions. This in turn makes these long strings display properly in the [Bindings index](https://docs.zephyrproject.org/latest/build/dts/api/bindings.html#devicetree-binding-index).

Do not use any other style for long or multi-line strings.

Reply:
thanks, I've fixed it in the latest push.